### DISCUSSION TITLE
Issues in cast call (Section 4: Airdrop and Signatures)

### DISCUSSION THREAD

**Discussion by farman1094** [Cyfrin/foundry-full-course-cu#2644] [Q&A] [answered]
Cast call is working for `Sepolia` but not for `Anvil`

![image](https://github.com/user-attachments/assets/636a094a-f5e6-44a5-ba54-a171c0fe7e74)

I have tried using 
ANVIL_RPC_URL=http://127.0.0.1:8545
and http://localhost:8545 
 
![image](https://github.com/user-attachments/assets/f56d83a2-126a-4cac-a189-f1c8bb205c86)

I have deployed the code to sepolia just to check this and same code is working fine. 
But how come it's working for the **Sepolia** and not for the **Anvil**

Note: (Some changes in my `foundry.toml` to keep in mind)
```toml
evm_version = "Cancun"
```

Reason behind is:

![image](https://github.com/user-attachments/assets/0254b5f6-d7cb-405b-bb08-bae7dcc1d6f2)

My code link: https://github.com/farman1094/Airdrop-Merkle
Thank you for your time and response.🙏

**Comment by EngrPips** [comment, marked as answer]
Hello @farman1094, You need to pass an `rpc-url` when deploying to `Anvil` if you want it to preserve the state of the blockchain. Or else it just runs the the process and disposes the data after returning them to you.

**Reply by EngrPips** [reply]
Did you figure it out already?

**Reply by farman1094** [reply]
Yeah! Thank you brother @EngrPips. At the time of deploying I was not using `rpc-url` to preserve the state of the blockchain.

**Reply by farman1094** [reply]
> Hello @farman1094, You need to pass an `rpc-url` when deploying to `Anvil` if you want it to preserve the state of the blockchain. Or else it just runs the the process and disposes the data after returning them to you.

I have still have some doubt's as well, sometimes its peserving the state sometime not. I have to use `--private-key` and `--broadcast` Any source of information where it's explained briefly. So I can learn about this?

**Reply by EngrPips** [reply]
I think this https://book.getfoundry.sh/reference/forge/forge-script might clarify things
